src/matlab2tikz.m

```diff
@@ -998,7 +998,9 @@ function addCustomCode(fid, before, code, after)
     backgroundColor = get(handle, 'Color');
     if ~isNone(backgroundColor)
         [m2t, col] = getColor(m2t, handle, backgroundColor, 'patch');
-        if ~strcmp(col, 'white')
+        % Fill background if not just white or if multiple axes and current
+        % is visible
+        if ~strcmp(col, 'white') || (numel(m2t.relevantAxesHandles) > 1 && isVisible(handle))
```

Comment:
Then maybe we should just leave out the `~strcmp(col, 'white')`, then the behavior is universal and one can probably set `set(gca,'Color','none')` or something to prevent that?